utils/build-windows.bat

```diff
@@ -0,0 +1,314 @@
+    -DLLVM_ENABLE_LIBXML2:BOOL=NO^
+    -DLLVM_ENABLE_ZLIB:BOOL=NO^
+    -DENABLE_X86_RELAX_RELOCATIONS:BOOL=YES^
+    -DLLVM_INSTALL_BINUTILS_SYMLINKS:BOOL=YES^
+    -DLLVM_INSTALL_TOOLCHAIN_ONLY:BOOL=YES^
+    -DLLVM_TOOLCHAIN_TOOLS:STRING="addr2line;ar;c++filt;dsymutil;dwp;llvm-ar;llvm-cov;llvm-cvtres;llvm-cxxfilt;llvm-dlltool;llvm-dwp;llvm-ranlib;llvm-lib;llvm-mt;llvm-nm;llvm-objdump;llvm-pdbutil;llvm-profdata;llvm-rc;llvm-readelf;llvm-readobj;llvm-size;llvm-strip;llvm-symbolizer;llvm-undname;nm;objcopy;objdump;ranlib;readelf;size;strings"^
+    -DCLANG_TOOLS="clang;clang-format;clang-headers;clang-tidy"^
+    -DCMAKE_CXX_FLAGS:STRING="/GS- /Oy"^
+    -DCMAKE_EXE_LINKER_FLAGS:STRING=/INCREMENTAL:NO^
+    -DCMAKE_SHARED_LINKER_FLAGS:STRING=/INCREMENTAL:NO %exitOnError%
+
+popd
+
+cmake --build "%build_root%\llvm" %exitOnError%
+cmake --build "%build_root%\llvm" --target install %exitOnError%
+
+goto :eof
+endlocal
+
+
+:build_cmark
+:: Configures and builds CMark
+setlocal enableextensions enabledelayedexpansion
+
+mkdir "%build_root%\cmark" %exitOnError%
+pushd "%build_root%\cmark"
+
+cmake "%source_root%\cmark"^
+    -G Ninja^
+    -DCMAKE_BUILD_TYPE=%CMAKE_BUILD_TYPE%^
+    -DCMAKE_C_COMPILER=cl^
+    -DCMAKE_CXX_COMPILER=cl^
+    -DCMAKE_CXX_FLAGS:STRING="/GS- /Oy"^
+    -DCMAKE_EXE_LINKER_FLAGS:STRING=/INCREMENTAL:NO^
+    -DCMAKE_SHARED_LINKER_FLAGS:STRING=/INCREMENTAL:NO %exitOnError%
+
+popd
+
+cmake --build "%build_root%\cmark" %exitOnError%
+
+goto :eof
+endlocal
+
+
+:build_swift
+:: Configures, builds, and installs Swift and the Swift Standard Library
+setlocal enableextensions enabledelayedexpansion
+
+mkdir "%build_root%\swift" %exitOnError%
+pushd "%build_root%\swift"
+
+cmake "%source_root%\swift"^
+    -G Ninja^
+    -DCMAKE_BUILD_TYPE=%CMAKE_BUILD_TYPE%^
+    -DCMAKE_C_COMPILER=cl^
+    -DCMAKE_CXX_COMPILER=cl^
+    -DCMAKE_INSTALL_PREFIX=%install_directory%^
+    -DClang_DIR=%build_root%\llvm\lib\cmake\clang^
+    -DSWIFT_PATH_TO_CMARK_BUILD=%build_root%\cmark^
+    -DSWIFT_PATH_TO_CMARK_SOURCE=%source_root%\cmark^
+    -DSWIFT_PATH_TO_LIBDISPATCH_SOURCE=%source_root%\swift-corelibs-libdispatch^
+    -DLLVM_DIR=%build_root%\llvm\lib\cmake\llvm^
+    -DSWIFT_INCLUDE_DOCS=NO^
+    -DSWIFT_WINDOWS_x86_64_ICU_UC_INCLUDE=%source_root%\icu-63_1\include\unicode^
+    -DSWIFT_WINDOWS_x86_64_ICU_UC=%source_root%\icu-63_1\lib64\icuuc.lib^
+    -DSWIFT_WINDOWS_x86_64_ICU_I18N_INCLUDE=%source_root%\icu-63_1\include^
+    -DSWIFT_WINDOWS_x86_64_ICU_I18N=%source_root%\icu-63_1\lib64\icuin.lib^
+    -DSWIFT_BUILD_STATIC_STDLIB=NO^
+    -DSWIFT_BUILD_STATIC_SDK_OVERLAY=NO^
+    -DLLVM_INSTALL_TOOLCHAIN_ONLY=YES^
+    -DSWIFT_BUILD_SOURCEKIT=YES^
+    -DSWIFT_ENABLE_SOURCEKIT_TESTS=NO^
+    -DSWIFT_INSTALL_COMPONENTS="autolink-driver;compiler;clang-resource-dir-symlink;stdlib;sdk-overlay;editor-integration;tools;sourcekit-inproc;swift-remote-mirror;swift-remote-mirror-headers"^
+    -DSWIFT_PARALLEL_LINK_JOBS=8^
```

Comment:
Might be nice to use: `wmic cpu get NumberOfCores` rather than `wmic cpu get NumberOfLogicalProcessors`

Reply:
This number was found by trial and error. The number of processors in AWS CPUs and the amount of RAM grow linearly, but seemed that the usage of the RAM while compiling wasn’t. Without this the compilation was failing when exhausting the RAM. We reduced it to try to keep under the limit (which works).

AWS machines do not differentiate between logical processors and cores, since they are virtual CPUs anyway.

Reply:
Ah, okay.  Could you add a comment indicating that the number is empirical and why it is mapped to a specific value currently.

Reply:
A comment will be added.